mesa/space.py

```diff
@@ -323,19 +322,19 @@ def _remove_agent(self, pos, agent):
         """ Remove the agent from the given location. """
         x, y = pos
         self.grid[x][y] = None
-        self.empties.append(pos)
+        self.empties.add(pos)
 
     def is_cell_empty(self, pos):
         """ Returns a bool of the contents of a cell. """
         x, y = pos
-        return True if self.grid[x][y] == self.default_val() else False
+        return self.grid[x][y] == self.default_val()
 
     def move_to_empty(self, agent):
         """ Moves agent to a random empty cell, vacating agent's old cell. """
         pos = agent.pos
         if len(self.empties) == 0:
             raise Exception("ERROR: No empty cells")
-        new_pos = agent.random.choice(self.empties)
+        new_pos = agent.random.choice(sorted(self.empties))
```

Comment:
Looks clunky, but items in a set have no predictable order in Python and simply getting the first element or changing it to an unordered list would not yield reproducible results)